### PR TITLE
[PR1] DEV-109 add WebsocketHandle

### PR DESCRIPTION
**Pull request by jmaralo**
The websocket handle unifies all the websocket connections into one by wrapping each message with a type and then sending the previous data with it.

The current implementation encapsulates all the previous components except the logger

### CODE REVIEW

File: src/excel_adapter/internals/excel_reader.go [hunk 1]
```diff
@@ -52,7 +49,7 @@ func parseSheet(name string, cols [][]string) models.Sheet {
 	tables := make(map[string]models.Table)
 
 	for name, bound := range findTables(cols) {
-		println(name)
+		fmt.Println(name)
```
Comment (lipezaballa) [inline src/excel_adapter/internals/excel_reader.go:52]: Delete prints

File: src/websocket_handle/models/message.go [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package models
+
+import "encoding/json"
+
+type Message struct {
+	Kind string          `json:"type"`
```
Comment (lipezaballa) [inline src/websocket_handle/models/message.go:6]: Kind instead of Type?

**Review by lipezaballa — APPROVED** [review]

File: src/connection_transfer/connection_transfer.go [hunk 3]
```diff
@@ -36,22 +41,21 @@ func (connectionTransfer *ConnectionTransfer) Update(name string, up bool) {
...
-func (ConnectionTransfer *ConnectionTransfer) Close() {
-	for _, socket := range ConnectionTransfer.sockets {
-		socket.Close()
+func (connectionTransfer *ConnectionTransfer) send() {
+	msg, err := json.Marshal(mapToArray(connectionTransfer.boardStatus))
+	if err != nil {
+		log.Printf("connectionTransfer: send: %s\n", err)
+		return
+	}
+	connectionTransfer.channel <- ws_models.MessageTarget{
```
Comment (smorsua) [inline src/connection_transfer/connection_transfer.go:53]: Create function that returns a `ws_models.MessageTarget` and use it here, in data transfer, etc.

File: src/websocket_handle/ws_handle.go [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+			case msg := <-output:
+				if len(msg.Target) == 0 {
+					for _, conn := range handle.conns {
+						err := conn.WriteJSON(msg.Msg)
+						if err != nil {
+							handle.Close(conn)
+						}
+					}
+				} else {
+					for _, target := range msg.Target {
```
Comment (smorsua) [inline src/websocket_handle/ws_handle.go:46]: What does the message target represent?

Comment (jmaralo) [reply src/websocket_handle/ws_handle.go:46]: Where did the message come from or for who the message is, this allows to send the same message to multiple clients or to answer a specific request

File: src/log_handle/log.go [hunk 5]
```diff
@@ -115,28 +122,32 @@ func (logger *LogHandle) createFile(value string) *os.File {
...
 
-func (logger *LogHandle) HandleRequest(w http.ResponseWriter, r *http.Request) {
-	defer r.Body.Close()
-	payload, err := io.ReadAll(r.Body)
-	if err != nil {
-		log.Fatalf("log handle: HandleRequest: %s\n", err)
-	}
+func (logger *LogHandle) listenWS() {
+	for msg := range logger.channel {
+		if logger.logSession == "" || msg.Target[0] == logger.logSession {
```
[the remark this answers is not in the document]
Comment (jmaralo) [reply src/log_handle/log.go:127]: nil is not defined for string, replacing "" with nil throws an error

**Review by smorsua — APPROVED** [review]

**Pull request merged**